src/main/java/com/google/sps/servlets/MarkTodoServlet.java

```diff
@@ -34,6 +34,9 @@
     public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
         HttpSession session = request.getSession();
         String profileID = (String)session.getAttribute("ProfileID");
+        if (profileID.equals(null)) {
```

Comment:
Have a suggestion here. Instead of returning nothing from the api, we could return a response like a status code 403. That way it wont break anywhere in the js